com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -889,27 +894,27 @@ public virtual void Heuristic(in ActionBuffers actionsOut)
             // Disable deprecation warnings so we can call the legacy overload.
 #pragma warning disable CS0618
 
+            Array.Clear(m_LegacyHeuristicCache, 0, m_LegacyHeuristicCache.Length);
```

Comment:
I think it's reasonable that the values are persistent across steps. Discussed changing this in https://jira.unity3d.com/browse/MLA-1129

Reply:
ok, i can take it out